pkg/identity/cache/allocator.go

```diff
@@ -273,7 +273,9 @@ func (m *CachingIdentityAllocator) Close() {
 
 	m.IdentityAllocator.Delete()
 	if m.events != nil {
-		close(m.events)
```

Comment:
nit: Since we don't close `m.events` anymore, could we make `type AllocatorEventChan chan AllocatorEvent` into `type AllocatorEventChan chan <-AllocatorEvent` to disallow any users from accidentally closing it?

Reply:
Hmm this is actually a bit more involved than I though - looking into other uses of that type.